pkg/controller/volume/attachdetach/util/util.go

```diff
@@ -249,3 +274,109 @@ func ProcessPodVolumes(pod *v1.Pod, addVolumes bool, desiredStateOfWorld cache.D
 	}
 	return
 }
+
+func translateInTreeSpecToCSIIfNeeded(spec *volume.Spec, nodeName types.NodeName, vpm *volume.VolumePluginMgr, csiMigratedPluginManager volume.CSIMigratedPluginManager, csiTranslator volume.InTreeToCSITranslator) (*volume.Spec, error) {
+	migratable, err := csiMigratedPluginManager.IsMigratable(spec)
+	if err != nil {
+		return nil, err
+	}
+	migrationSupportedOnNode, err := isCSIMigrationSupportedOnNode(nodeName, spec, vpm, csiMigratedPluginManager)
+	if err != nil {
+		return nil, err
+	}
+	if migratable && migrationSupportedOnNode {
+		spec, err = volume.TranslateInTreeSpecToCSI(spec, csiTranslator)
```

Comment:
nit: I would prefer if this was more explicitly *not* the input variable. I think the behavior is the same though since we do no in-place modification?

Reply:
Since these are pointers, it does not make much of a difference. I have changed the names to be cleared as suggested